### ISSUE TITLE
`cylc config --list-platforms`

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-flow#4557]
Users at sites with pre-set platforms will want to query the list of platforms and platform groups.

Using `cylc config -i [platforms]; cylc config -i [platform groups]` is a bit clunky, and may overwhelm some users with the quantity of more esoteric settings (ssh commands &c) returned.

This ticket proposes the addition of the option `--list-platforms` to `cylc config`. This option should return a list of platforms and platform groups, with the following items from the config:
- Platform regex (it's "name" in the global config)
- Hosts list / Platform method
- Job Runner
- Any fields defined under meta

**questions**
- Are there other fields we consider essential, such as "selection method"?
- Do we want to allow `cylc config --platform <name>` to give a preview of what will happen if you use <name> in a workflow? 

E.g 
```
<name> matches platform <platform> and will be run on a host in {hosts} using {batch system}

<name> matches platform group <group name> and will be run on a platform from the following list:
  - foo and will be run on a host in {hosts} using {batch **system}`**
```

**Additional context**
Discussed in an [element conversation](https://matrix.to/#/!hwZqSYihGPuhDdIzIP:matrix.org/$1639746821549588VCnCm:matrix.org?via=matrix.org)

**Pull requests welcome!**

**Comment by hjoliver** [comment]
Agreed this is a good idea.

However, I suggest `cylc config --describe-platforms` or similar, for giving platform (and group) names (or regexes) plus metadata. `--list-platforms` sounds like it would list all platform config (which of course can be done with `config -i`).

**Comment by wxtim** [comment]
> 

    cylc config --platform-names # just the names
    cylc config --platform-meta # names + metadata


I like this.

Don't forget that I want this CLI to access both platforms and platform groups, because when a user sets `[runtime][TASK]platform=foo` we don't know whether foo is a platform or a group.

Do we want  to reverse the config order or document that the selection process goes from the bottom up -i.e if 

```ini
[platforms]
    [[foo]]
        hosts = foo
    [[fo.]]
        hosts = bar
[platform groups]
    [[f.*]]
        hosts = baz
```

then the example of `platform=foo` will run on host `baz`?

**Comment by wxtim** [comment]
### Proposal

Add the following options to `cylc config`:
- `--platform-names`, which prints out a list of platform and platform group regexes.
- `--platform-meta` which prints out:
  - regexes
  - metadata
  - hosts
  - job runner
  - install target
- In both cases the list should be prepended by a note explaining that
  -  Cylc will search the list from the bottom.
  -  Regex definitions create distinct platforms for each possible match.
- Create further (post 8.0.0) issues to discuss the possibility of
  - A `platform=foo` simulator, which will run the platform logic in a standalone way to give users the ability to discover what platform their workflow setting will select.
  - A checker for unselectable platforms where a platform/group high on the list is masked by a regex below it (checked earlier by Cylc).

**Comment by hjoliver** [comment]
> >    --platform-names --platform-meta
> 
> I don't think there's a use case for two commands.

I kinda disagree. Most of the time most users won't care about platform config details, which should have been taken care of for them by site config, but they might occasionally need a reminder of what the Cylc platforms are called (typically they only thing they need to know, for use in their workflow configs).

This is a obviously not a big deal because it's not that hard to parse the name from the full platform config, but if it's convenient then why not?  There's not much overhead for us in supporting two command options (not "two commands").

**Comment by oliver-sanders** [comment]
Fair'nough.

Perhaps it would be useful for `cylc config` to have something like a `--depth` argument for this sort of thing.
